File: Rivington.IG.Web/Rivington.IG.Domain.Models/OrderManagement/InspectionOrderProperty/HomeDetail.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Rivington.IG.Domain.Models.OrderManagement
{
    public class InspectionOrderPropertyHomeDetail
    {
        [Parent]
        public InspectionOrderProperty InspectionOrderProperty { get; set; }
        [Key]
        public Guid Id { get; set; }
        
        public string ArchitecturalStyleId { get; set; }
        [ForeignKey(nameof(ArchitecturalStyleId))]
        public ArchitecturalStyle ArchitecturalStyle { get; set; }
        
        public string FramingTypeId { get; set; }
        [ForeignKey(nameof(FramingTypeId))]
        public FramingType FramingType { get; set; }
        
        public string ConstructionQualityId { get; set; }
        [ForeignKey(nameof(ConstructionQualityId))]
        public ConstructionQuality ConstructionQuality { get; set; }
        
        public string HomeShapeId { get; set; }
        [ForeignKey(nameof(HomeShapeId))]
        public HomeShape HomeShape { get; set; }
        
        public string PrimaryExteriorWallCoveringId { get; set; }
        [ForeignKey(nameof(PrimaryExteriorWallCoveringId))]
        public PrimaryExteriorWallCovering PrimaryExteriorWallCovering { get; set; }
        
        public string SecondaryExteriorWallCoveringId { get; set; }
        [ForeignKey(nameof(SecondaryExteriorWallCoveringId))]
        public SecondaryExteriorWallCovering SecondaryExteriorWallCovering { get; set; }
        
        public string PrimaryRoofCoveringId { get; set; }
        [ForeignKey(nameof(PrimaryRoofCoveringId))]
        public PrimaryRoofCovering PrimaryRoofCovering { get; set; }
        
        public string SecondaryRoofCoveringId { get; set; }
        [ForeignKey(nameof(SecondaryRoofCoveringId))]
        public SecondaryRoofCovering SecondaryRoofCovering { get; set; }

        public string RoofTypeId { get; set; }
        [ForeignKey(nameof(RoofTypeId))]
        public RoofType RoofType { get; set; }
        public string RoofShapeId { get; set; }
        [ForeignKey(nameof(RoofShapeId))]
        public RoofShape RoofShape { get; set; }

        public string RoofPitchId { get; set; }
        [ForeignKey(nameof(RoofPitchId))]
        public RoofPitch RoofPitch { get; set; }

        public string FoundationTypeId { get; set; }
        [ForeignKey(nameof(FoundationTypeId))]
        public FoundationType FoundationType { get; set; }
        
        public int? NumberofStories { get; set; }
        
        public string SlopeOfSiteId { get; set; }
        [ForeignKey(nameof(SlopeOfSiteId))]
        public SlopeOfSite SlopeOfSite { get; set; }

        public ICollection<InspectionOrderPropertyHomeDetailLocales> Locales { get; set; }

        public static void BuildModel(ModelBuilder modelBuilder)
        {
            var entTypeName = nameof(InspectionOrderPropertyHomeDetail);
            modelBuilder.Entity<InspectionOrderProperty>()
                .HasOne(a => a.HomeDetail).WithOne(b => b.InspectionOrderProperty)
                .HasForeignKey<InspectionOrderPropertyHomeDetail>()
                .HasConstraintName($"FK_{nameof(InspectionOrderProperty)}_{entTypeName}")
                .OnDelete(DeleteBehavior.Cascade);

            var entBuilder = modelBuilder.Entity<InspectionOrderPropertyHomeDetail>();
            entBuilder.HasOne(a => a.ArchitecturalStyle).WithMany().HasConstraintName($"FK_{entTypeName}_{nameof(ArchitecturalStyle)}");
            entBuilder.HasOne(a => a.FramingType).WithMany().HasConstraintName($"FK_{entTypeName}_{nameof(FramingType)}");
            entBuilder.HasOne(a => a.ConstructionQuality).WithMany().HasConstraintName($"FK_{entTypeName}_{nameof(ConstructionQuality)}");
            entBuilder.HasOne(a => a.HomeShape).WithMany().HasConstraintName($"FK_{entTypeName}_{nameof(HomeShape)}");
            entBuilder.HasOne(a => a.PrimaryExteriorWallCovering).WithMany().HasConstraintName($"FK_{entTypeName}_{nameof(PrimaryExteriorWallCovering)}");
            entBuilder.HasOne(a => a.SecondaryExteriorWallCovering).WithMany().HasConstraintName($"FK_{entTypeName}_{nameof(SecondaryExteriorWallCovering)}");
            entBuilder.HasOne(a => a.PrimaryRoofCovering).WithMany().HasConstraintName($"FK_{entTypeName}_{nameof(PrimaryRoofCovering)}");
            entBuilder.HasOne(a => a.SecondaryRoofCovering).WithMany().HasConstraintName($"FK_{entTypeName}_{nameof(SecondaryRoofCovering)}");
            entBuilder.HasOne(a => a.RoofType).WithMany().HasConstraintName($"FK_{entTypeName}_{nameof(RoofType)}");
            entBuilder.HasOne(a => a.RoofShape).WithMany().HasConstraintName($"FK_{entTypeName}_{nameof(RoofShape)}");
            entBuilder.HasOne(a => a.RoofPitch).WithMany().HasConstraintName($"FK_{entTypeName}_{nameof(RoofPitch)}");
            entBuilder.HasOne(a => a.FoundationType).WithMany().HasConstraintName($"FK_{entTypeName}_{nameof(FoundationType)}");
            entBuilder.HasOne(a => a.SlopeOfSite).WithMany().HasConstraintName($"FK_{entTypeName}_{nameof(SlopeOfSite)}");

            InspectionOrderPropertyHomeDetailLocales.BuildModel(modelBuilder);
        }
    }
}

File: Rivington.IG.Web/Rivington.IG.API/src/app/modules/shared/models/ordermanagement/inspection-order/checklist/property/home-detail.ts
import { InspectionOrderProperty } from "../property";
import { InspectionOrderPropertyHomeDetailLocales } from "./home-detail/InspectionOrderPropertyHomeDetailLocales";
import { ArchitecturalStyle } from "./home-detail/architectural-style";
import { ConstructionQuality } from "./home-detail/construction-quality";
import { FoundationType } from "./home-detail/foundation-type";
import { FramingType } from "./home-detail/framing-type";
import { HomeShape } from "./home-detail/home-shape";
import { PrimaryExteriorWallCovering } from "./home-detail/primary-exterior-wall-covering";
import { PrimaryRoofCovering } from "./home-detail/primary-roof-covering";
import { RoofType } from "./home-detail/roof-type";
import { SecondaryExteriorWallCovering } from "./home-detail/secondary-exterior-wall-covering";
import { SecondaryRoofCovering } from "./home-detail/secondary-roof-covering";
import { SlopeOfSite } from "./home-detail/slope-of-site";
import { RoofShape } from "./home-detail/roof-shape";
import { RoofPitch } from "./home-detail/roof-pitch";

export class InspectionOrderPropertyHomeDetail {
    /*[Parent]*/
    inspectionOrderProperty?: InspectionOrderProperty;
    id?: string;
    architecturalStyleId?: string;
    /*[ForeignKey(nameof(ArchitecturalStyleId))]*/
    architecturalStyle?: ArchitecturalStyle;
    framingTypeId?: string;
    /*[ForeignKey(nameof(FramingTypeId))]*/
    framingType?: FramingType;
    constructionQualityId?: string;
    /*[ForeignKey(nameof(ConstructionQualityId))]*/
    constructionQuality?: ConstructionQuality;
    homeShapeId?: string;
    /*[ForeignKey(nameof(HomeShapeId))]*/
    homeShape?: HomeShape;
    primaryExteriorWallCoveringId?: string;
    /*[ForeignKey(nameof(PrimaryExteriorWallCoveringId))]*/
    primaryExteriorWallCovering?: PrimaryExteriorWallCovering;
    secondaryExteriorWallCoveringId?: string;
    /*[ForeignKey(nameof(SecondaryExteriorWallCoveringId))]*/
    secondaryExteriorWallCovering?: SecondaryExteriorWallCovering;
    primaryRoofCoveringId?: string;
    /*[ForeignKey(nameof(PrimaryRoofCoveringId))]*/
    primaryRoofCovering?: PrimaryRoofCovering;
    secondaryRoofCoveringId?: string;
    /*[ForeignKey(nameof(SecondaryRoofCoveringId))]*/
    secondaryRoofCovering?: SecondaryRoofCovering;
    roofShapeId?: string;
    /*[ForeignKey(nameof(roofShapeId))]*/
    roofShape?: RoofShape;
    roofPitchId?: string;
    /*[ForeignKey(nameof(roofPitchId))]*/
    roofPitch?: RoofPitch;
    foundationTypeId?: string;
    /*[ForeignKey(nameof(FoundationTypeId))]*/
    foundationType?: FoundationType;
    numberofStories?: number;
    slopeOfSiteId?: string;
    /*[ForeignKey(nameof(SlopeOfSiteId))]*/
    slopeOfSite?: SlopeOfSite;
    locales?: InspectionOrderPropertyHomeDetailLocales[];
    
    public constructor(init?:Partial<InspectionOrderPropertyHomeDetail>) {
        Object.assign(this, init);
    }
}